On Windows path separator (\) needs to be escaped (\\) or better replaced with forward slash

See sample repo https://github.com/manwithafox/esbuild-dirname-issue-on-windows/tree/main
```
C:\dev\esbuild-dirname-issue-on-windows (master)>npm run package

> esbuild-dirname-issue-on-windows@1.0.0 package
> node build

X [ERROR] Syntax error "t"

    src/util/index.ts:3:111:
      3 │           filename: "C:\dev\esbuild-dirname-issue-on-windows\src\util\index.ts",
        ╵                                                                   ^

C:\dev\esbuild-dirname-issue-on-windows (master)>
```
Sample only fails because the path contains a `util` folder which then gets rendered as **\u**til without 4-digit unicode following. Else it would live on undiscovered.
Backslash path separator needs to be escaped or better replaced by forward slash, matching the relative path.
Expected output of sample repo would be:

```
// src/util/index.ts
var __fileloc = {
  filename: "C:/dev/esbuild-dirname-issue-on-windows/src/util/index.ts",
  dirname: "C:/dev/esbuild-dirname-issue-on-windows/src/util",
  relativefilename: "src/util/index.ts",
  relativedirname: "src/util"
};
var utilFilename = __fileloc.filename;

// src/index.ts
console.log(utilFilename);
```

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,5 +1,6 @@
 import { Plugin, Loader } from "esbuild";
 import path from "path";
+import process from "process";
 import fs from "fs-extra";
 
 interface FilelocPluginOptions {
@@ -14,12 +15,14 @@ export const filelocPlugin = (options?: FilelocPluginOptions): Plugin => ({
     build.onLoad(
       { filter: /.\.(js|ts|jsx|tsx)$/, namespace: "file" },
       async (args) => {
+        const isWindows = /^win/.test(process.platform);
+        const esc = p => isWindows ? p.replace(/\\/g, '/') : p;
         const variables = `
         const __fileloc = {
-          filename: "${args.path}",
-          dirname: "${path.dirname(args.path)}",
-          relativefilename: "${path.relative(rootDir, args.path)}",
-          relativedirname: "${path.relative(rootDir, path.dirname(args.path))}"
+          filename: "${esc(args.path)}",
+          dirname: "${esc(path.dirname(args.path))}",
+          relativefilename: "${esc(path.relative(rootDir, args.path))}",
+          relativedirname: "${esc(path.relative(rootDir, path.dirname(args.path)))}"
         };
         let __line = 0;
       `;

